bin/build-package-containing-file.js

```diff
@@ -0,0 +1,39 @@
+const fs = require( 'fs' );
+const { spawnSync } = require( 'child_process' );
+
+const args = process.argv.slice( 2 );
+if ( args.length < 1 ) {
+	console.error( 'failed to rebuild package: at least one filename must be provided' );
+	process.exit( 1 );
+}
+const fileName = args[ 0 ];
+if ( ! fileName ) {
+	console.error( 'failed to rebuild package: missing filename' );
+	process.exit( 1 );
+}
+const packageDirectoryMatches = fileName.match( /^packages\/[^/]+/ );
+if ( packageDirectoryMatches.length < 1 ) {
+	console.error( 'failed to rebuild package: missing package directory' );
+	process.exit( 1 );
+}
+const packageDirectory = packageDirectoryMatches[ 0 ];
+if ( ! packageDirectory ) {
+	console.error( 'failed to rebuild package: missing package directory contents' );
+	process.exit( 1 );
+}
```

Comment:
This would fail on Windows (because Windows uses \ not /).. I would do this:

On top of the script, I'd store the packages absolute dir, 

```js
const packagesFolder = path.resolve( '..', 'packages' );
```

Then below I'd do:

```js
// relative path between packages folder and the affected file
// if you modify data-stores/tsconfig.json, the relative path will be
// data-stores/tsconfig.json 
const affectedFileRelativePath = path.relative( packagesFolder, changedFile );

if(affectedFileRelativePath.beginsWith('..')) {
	console.error( 'Path is outside packages directory' );
    process.exit( 1 );
}

// notice using path.sep, because it's OS agnostic. 
const affectedPackage = relativePath.split( path.sep )[ 0 ];
```